src/Persistance/RepositoryImplementations/SystemRegisterRepository.cs

```diff
@@ -228,8 +225,52 @@ private static ValueTask<RegisteredSystem> ConvertFromReaderToSystemRegister(Npg
     }
 
     /// <inheritdoc/> 
-    public Task<bool> CreateClient(string clientId)
+    public async Task<bool> CreateClient(string clientId)
     {
-        throw new NotImplementedException();
+        Guid insertedId = Guid.Parse(clientId);
+
+        const string QUERY = /*strpsql*/@"
+            INSERT INTO altinn_authentication.maskinporten_client(
+            client_id)
+            VALUES
+            (@new_client_id)";
+
+        try
+        {
+            await using NpgsqlCommand command = _datasource.CreateCommand(QUERY);
+            command.Parameters.AddWithValue("new_client_id", insertedId);
+            return await command.ExecuteNonQueryAsync() > 0;
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Authentication // SystemRegisterRepository // CreateClient // Exception");
+            throw;
+        }
+    }
+
+    /// <inheritdoc/> 
+    public async Task<Guid?> RetrieveGuidFromStringId(string id)
+    {
+        const string GUIDQUERY = /*strpsql*/@"
+                SELECT hidden_internal_id
+                FROM altinn_authentication.system_register
+        	    WHERE altinn_authentication.system_register.registered_system_id = @registered_system_id;
+                ";
+    
+        try
+        {
+            await using NpgsqlCommand guidCommand = _datasource.CreateCommand(GUIDQUERY);
+
+            guidCommand.Parameters.AddWithValue("registered_system_id", id);
+
+            return await guidCommand.ExecuteEnumerableAsync()
+                .SelectAwait(NpgSqlExtensions.ConvertFromReaderToGuid)
+                .SingleOrDefaultAsync();
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "Authentication // SystemRegisterRepository // CreateClient // Exception");
```

Comment:
comment to be updated to match the action

Reply:
ok fixed